### PR TITLE
Rm incorporate draper

### PR DESCRIPTION
**Pull request by rmathew1992**
This PR adds a feature that allows users to see whether or not a song is popular or not. 

A plays column has been added to the songs table that keeps tracks of how many times a song has been played. 

A decorator was added to the songs table. This decorator has a method 'popularity', if a song has been played more then a requisite number of times it is considered to be popular and a message stating that it is popular is attached to the song and attached to the JSON object. 

_Sample Request_

```
GET /songs/8689be0d-b476-49b4-9b16-4a44add8c543 HTTP/1.1
Accept: application/vnd.spotify-apprentice-app.com; version=1
```

_Sample Response_

```
{
  "songs": [
    {
      "id": "8689be0d-b476-49b4-9b16-4a44add8c543",
      "name": "Dirty Diana",
      "spotify_uri": "90db99e058ef45a93902",
      "popularity": "Dirty Diana has 3001 plays, it is popular",
      "artist_id": "a0e234ce-7226-4ba6-885b-12a5fe8489df"
    }
  ],
  "artists": [
    {
      "id": "a0e234ce-7226-4ba6-885b-12a5fe8489df",
      "name": "Michael Jackson",
      "description": "The king of pop",
      "spotify_uri": "9be4f808260485c56f62",
      "songs_count": 0,
      "song_ids": [
        "8689be0d-b476-49b4-9b16-4a44add8c543",
        "65bd5a47-ad68-4bd5-a6b1-b469ec3baa4a",
        "20eec79f-2892-4a2b-b8c5-7e403ed3c1d6"
      ]
    }
  ]
}
```

### CODE REVIEW

File: app/decorator/song_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class SongDecorator < Draper::Decorator
+  def popularity
+    if object.plays > 3000
+      "#{object.name} has #{object.plays} plays, it is popular"
```
Comment (dthyresson) [inline app/decorator/song_decorator.rb]: Ah, so -- this won't happen due to your `> 3000` rule, but what if it had just one play.

How could you use this decorator to say "has 1 play"?

Hint: https://github.com/drapergem/draper#accessing-helpers

Comment (dthyresson) [inline app/decorator/song_decorator.rb]: Say, as an `else` that says, " ... only have 1 play" or ".. just 100 plays"

File: spec/factories/songs.rb [hunk 2]
```diff
@@ -3,5 +3,6 @@
     name 'Dirty Diana'
     artist
     spotify_uri { SecureRandom.hex(10) }
+    plays 3001
```
Comment (dthyresson) [inline spec/factories/songs.rb:6]: So, this is a great sport for a feature of FactoryGirl called `traits`.

You can create a "popular" song and an "unpopular" song as traits.

The use the factory and traits in your testcases.

See: https://github.com/thoughtbot/factory_girl/blob/master/GETTING_STARTED.md#traits

Comment (dthyresson) [inline spec/factories/songs.rb:6]: The you can have testcases for popular songs, unpopular songs, almost popular ones, songs without and plays, etc really easily.

File: app/models/song.rb [hunk 3]
```diff
@@ -4,6 +4,7 @@ class Song < ActiveRecord::Base
   validates :name, presence: true
   validates :spotify_uri, presence: true, uniqueness: true
   validates :artist_id, presence: true
+  validates :plays, presence: true
```
Comment (dthyresson) [inline app/models/song.rb:7]: How are some ways you could make sure that a Song doesn't have -10 plays?

File: Gemfile [hunk 4]
```diff
@@ -28,6 +28,7 @@ gem 'uglifier'
 gem 'unicorn'
 gem 'versionist'
 gem 'counter_culture', '~> 0.1.23'
+gem 'draper', '~>1.3'
```
Comment (dthyresson) [inline Gemfile]: This one's kinda nitpicky, but we try to keep the gems in alpha-order.

That's so we know that we shouldn't add the gem twice -- or prevent us from adding it twice b/c we don't have to scan the entire list to see if it is already listed.

File: db/schema.rb [hunk 5]
```diff
@@ -50,6 +50,7 @@
     t.string   "name"
     t.string   "spotify_uri"
     t.uuid     "artist_id"
+    t.integer  "popularity"
```
Comment (dthyresson) [inline db/schema.rb]: Wait, do we need/use this db column attribute?

**Comment by rmathew1992** [comment]
@dthyresson What do you think?

File: app/models/song.rb (hunk 3 above)
Comment (rmathew1992) [inline app/models/song.rb:7]: @dthyresson Rails has a :greater_than_or_equal_to validator

**Pull request merged**